// simple-weather/lib/v2.5/current.js
"use strict";

var config = require("../../config")(),
    api = require("../api");

module.exports = {
    byCityName: byCityName,
    byCityId: byCityId,
    byCityIds: byCityIds,
    byGeoLocation: byGeoLocation,
    byZipcode: byZipcode
};

//api.openweathermap.org/data/2.5/weather?q={city name}
//api.openweathermap.org/data/2.5/weather?q={city name},{country code}
function byCityName(city, country) {
    var q = city;

    if (country) {
        q += ',' + country
    }

    return api.request({
        method: "GET",
        url: config.api["v2.5"].url + '/weather',
        query: {
            q: q
        }
    });
}

//api.openweathermap.org/data/2.5/weather?id=2172797
function byCityId(cityId) {
    return api.request({
        method: "GET",
        url: config.api["v2.5"].url + '/weather',
        query: {
            id: cityId
        }
    });
}

//http://api.openweathermap.org/data/2.5/group?id=524901,703448,2643743
function byCityIds(cityIds) {
    return api.request({
        method: "GET",
        url: config.api["v2.5"].url + '/group',
        query: {
            id: cityIds.join(',')
        }
    });
}

//api.openweathermap.org/data/2.5/weather?lat=35&lon=139
function byGeoLocation(latitude, longitude) {
    return api.request({
        method: "GET",
        url: config.api["v2.5"].url + '/weather',
        query: {
            lat: latitude,
            lon: longitude
        }
    });
}

//api.openweathermap.org/data/2.5/weather?zip=94040,us
function byZipcode(zipcode, country) {
    var q = zipcode;

    if (country) {
        q += ',' + country
    }

    return api.request({
        method: "GET",
        url: config.api["v2.5"].url + '/weather',
        query: {
            zip: q
        }
    });
}
